### PR TITLE
Issue 347: wrong dialect detection

### LINKED ISSUE #347: Wrong CSV parsing
Last staging of all.

subset of CSV:
```
PronounceGIF
"With a hard ""g,"" like ""gift"""
```
On upload this CSV i get error here: https://github.com/mindsdb/mindsdb_native/blob/stable/mindsdb_native/libs/data_sources/file_ds.py#L79
reason is `len(file_list_data) != len(col_map)`
```
len(file_list_data[1]) == 2  # ['"With a hard ""g', '"" like ""gift"""']
len(col_map) = 1
```
Full CSV:
[Cleaned.zip](https://github.com/mindsdb/mindsdb_native/files/5666230/Cleaned.zip)

### PR DESCRIPTION
**Pull request by ilia-tsyplenkov**
Closes #347 
- **What**
    - there is a data parsing error if column contains data in format:

             ```
            PronounceGIF
            "With a hard ""g,"" like ""gift"""
            ```

- **Why**
   - Root cause is that csv.Sniffer that used to deduce data dialect works with headers row (not data row) as template

- **How**
   - data row provided to csv.Sniffer as template if possible

### CODE REVIEW

File: mindsdb_native/libs/data_sources/file_ds.py [hunk 1]
```diff
@@ -186,16 +188,22 @@ def _getDataIo(self, file):
             data.seek(0)
             first_few_lines = []
             i = 0
+
+            # need to have some templates to deduce a dialect
```
Comment (George3d6) [inline mindsdb_native/libs/data_sources/file_ds.py]: What do you mean by "templates" here and bellow ? Could you give some examples.

File: mindsdb_native/libs/data_sources/file_ds.py [hunk 2]
```diff
@@ -186,16 +188,22 @@ def _getDataIo(self, file):
...
+            # but it is not a good idea to deduce dialect by header row
+            # data[0]
             for line in data:
                 if line in ['\r\n','\n']:
                     continue
```
Comment (George3d6) [inline mindsdb_native/libs/data_sources/file_ds.py]: Not added by this PR, but not exactly sure why this code has a "continue" it should have a break, i.e. this indicates a final line that only contains a newline.

File: mindsdb_native/libs/data_sources/file_ds.py [hunk 3]
```diff
@@ -186,16 +188,22 @@ def _getDataIo(self, file):
...
                 first_few_lines.append(line)
                 i += 1
```
Comment (George3d6) [inline mindsdb_native/libs/data_sources/file_ds.py]: Again, was here before, not introduced by this pr, but much cleaener to do `for i, line in enumerate(data)` than to have a manual incrementing `i`.

**Review by George3d6 — CHANGES_REQUESTED** [review]
Please add an automated test for the failing data format to be able to confirm this is working and that we won't break the detection in the future.

File: mindsdb_native/libs/data_sources/file_ds.py (hunk 3 above)
Comment (ilia-tsyplenkov) [reply mindsdb_native/libs/data_sources/file_ds.py]: totally agree with this note. Done

File: mindsdb_native/libs/data_sources/file_ds.py (hunk 2 above)
Comment (ilia-tsyplenkov) [reply mindsdb_native/libs/data_sources/file_ds.py]: You are right, according to RFC spec CSV file couldn't contain empty lines in the middle. Fixed

File: mindsdb_native/libs/data_sources/file_ds.py (hunk 1 above)
Comment (ilia-tsyplenkov) [reply mindsdb_native/libs/data_sources/file_ds.py]: under template I mean `sample`
```
class csv.Sniffer
The Sniffer class is used to deduce the format of a CSV file.

The Sniffer class provides two methods:

sniff(sample, delimiters=None)¶
....
```

**Pull request merged**